src/Umbraco.Core/PropertyEditors/BlockGridConfiguration.cs

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Umbraco.
+// See LICENSE for more details.
+
+using System.Runtime.Serialization;
+
+namespace Umbraco.Cms.Core.PropertyEditors
+{
+    /// <summary>
+    /// The configuration object for the Block Grid editor
+    /// </summary>
+    public class BlockGridConfiguration
+    {
+        [ConfigurationField("blocks", "Blocks", "views/propertyeditors/blockgrid/prevalue/blockgrid.blockconfiguration.html", Description = "Define Blocks based on Element Types.")]
+        public BlockGridBlockConfiguration[] Blocks { get; set; }  = Array.Empty<BlockGridBlockConfiguration>();
+
+        [DataContract]
+        public class BlockGridBlockConfiguration : IBlockConfiguration
+        {
+
+            [DataMember(Name ="columnSpanOptions")]
+            public BlockGridColumnSpanOption[] ColumnSpanOptions { get; set; }  = Array.Empty<BlockGridColumnSpanOption>();
+            [DataContract]
+            public class BlockGridColumnSpanOption
+            {
+                [DataMember(Name ="columnSpan")]
+                public int? ColumnSpan { get; set; }
+            }
```

Comment:
We probably should not be triple nesting public classes, are there a reason they cannot live in their own files? 
IE. Do we already have existing classes with this same ? 🤔 
I know these are probably only used by the `BlockGridConfiguration`, but they should probably live in their own file, but in a `BlockConfiguration` folder

Reply:
@Zeegaan this is following the general approach in the property editor space. However triple nesting is rather grim, and the outer class is a mess of properties and class declarations all entwined, so I have

- Moved triple nested classes to be nested directly under `BlockGridConfiguration`
- Reordered `BlockGridConfiguration` to have property declarations first and nested classes last